### DISCUSSION TITLE
Is there a way to undo my mistake configuring temperature probes?

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#883] [Q&A] [answered]
I am cannibalizing  an ET to build a Nixie & thought I was saving myself some hassle by configuring as much of the system as  possible on the bench before moving the box to the pad.  To that end, I  was trying to configure temperature probes, relays, etc. but I think I goofed & am not sure how to recover.

I am using the existing Pentair air and water temperature probes & other equipment.  I set the temperature probes up as 10K temperature probes & generic devices to feed the MegaBas several weeks ago & this morning finally moved the box to the pad & powered it up (from the ET's 24V AC using a bridge rectifier to get DC).  The good news is the Pi stack fired up, I have Ethernet connectivity to the Nixie, REM, and the Sequent and Atlas Scientific Hats, and am ready to start migrating the low voltage connections to the Nixie box.  Big celebration for me :-)

Like @CodeSlinger69 I wanted to start with the temperature probes & go from there.  However, when I added the probe wiring to the Nixie, no temperature data was available on the generic device tabs or in Nixie (dashPanel).  I believe I set the MegaBas input correctly--they're DIP switches on my board instead of jumpers.

I went back to check & edit if needed the temperature probes' feeds or generic device settings.  Although I thought I had configured them as shown in the wiki a few weeks ago, the input was not resistance and the units were not kohms.  Instead, it showed up as Raw Value--strange.  I changed them to match the wiki settings, but the GUI did not provide an option to save the device--in fact, no save or cancel buttons were present in the dialog.  Furthermore, I have not found a way to delete the probes or generic device so I could start afresh.  

So at this point, my questions are can I edit or delete the probes and/or generic devices via the GUI?  Do I need to edit anything on the MegaBas before deleting the generic device and/or its feed first?  Do I need to edit any configuration files to fix things?  Or, have I messed up to the point that I need to start fresh on the Pi?

Hoping there's a simple fix I just have yet to discover.

Thanks for your help!

**Comment by rstrouse** [comment]
If you are using a MEGA-BAS these should be configured as 10k sensors then fed to the generic device.  On the BAS you just enable or disable the channel.  Conversely, you can just select the type from the dropdown but it should match the jumpers on the board.  If you have dip switches then you do not have a MEGA-BAS.

**Reply by FlaMike** [reply]
Sequent apparently came out with version 4.2 of the Mega-Bas & that's what they shipped to me in December.  Their customer service folks responded to my question saying they had not updated the documentation on it & it's operationally the same as version 4.1. 

As far as I can tell, the only difference on the unit is the replacement of the jumper blocks with DIP switches.

Nevertheless, the problem I am having with my configuration is the generic settings dialog box does not allow me to save edits to the generic device.  So, I remain at a standstill.

**Reply by rstrouse** [reply]
When you click on the device in the list it doesn't open up the ability to save or delete?

**Comment by FlaMike** [comment]
Yes, that's correct.  I can change its settings, but save & delete are not
options.  So when I go from generic to I2C & back the device is unchanged.

**Comment by FlaMike** [comment]
BTW, if I add a device I can change or delete it.  Just don't seem to be
able to modify or delete the two "existing" temp probes.

**Comment by FlaMike** [comment]
P. P. S. The buttons are not greyed out, they don't appear in the dialog
box.  I can screenshot tomorrow, if you wish.

**Comment by FlaMike** [comment, marked as answer]
Spent some frustrating but productive time on this issue this morning.  To make a long story short, the lack of documentation from Sequent on version 4.2 of the Mega-Bas led me astray.  I assumed, incorrectly as it turns out, the DIP switches on the board corresponded to the jumper settings in the version 4.1 documentation.  It does not!

For the benefit of others who may get the version 4.2 Mega-Bas, the three DIP switches for each input must be set to OFF OFF ON as you look at the board from the input block (left) edge if you are using 10K temperature thermisters like those from Pentair and in the Wiki.  I mistakenly set the DIP switches to ON OFF OFF to match the 4.1 jumper documentation.  Reproduced below.

At least part of my misunderstanding is due to the orientation of the jumpers verses the DIP switches.  I believe the jumper blocks are perpendicular to the DIP switch orientation and I missed that difference completely.  It looks to me like the DIP switch settings are 180 degrees off of the drawing below.  I have yet to play with a flow switch at this time, but I am hoping its DIP switch setting is ON OFF OFF, again from the input block perspective.  More on that part of the story in a month or two when the DIY REM CHEM project is in full swing...

If you're using 1K thermister temperature probes, you're in luck...  I believe the correct DIP switch setting is OFF ON OFF.  Don't ask me how I found that out.

Anyway, after getting the DIP switches set correctly for the temperature probes, I was able to update the temperature probe configuration and get reasonable temperature readings in REM and on the dashPanel.  The missing Save and Delete Device buttons magically reappeared in REM once the temperature probe input settings were correct.  And for the record, when the settings were wrong, the configuration for the temperature probes changed behind my back & were not editable.  Oh, what a world!  Probably a good design (safe) feature by @rstrouse and @tagyoureit.

Whew!!!!  Now onto fun with RS-485, relays, and valves (once the cold and rain go away).  Freeze protect will probably get a workout soon, but at least we ain't getting snow (I hope!).

For the record, still running the ET since the temperature probes slowed my progress and I can't have half of a system given the weather forecast.  Getting ready sprint to the finish soon, however :-)


![image](https://github.com/tagyoureit/nodejs-poolController/assets/41341956/1c546a4a-ef72-4d94-88fc-a23b4c1b17bd)
